pkg/cloudfabric-controller/controllerframework/controller_framework_test.go

```diff
@@ -290,7 +290,338 @@ func TestConsolidateControllerInstances_ReturnValues_MergeAndAutoExtends(t *test
 	assert.Equal(t, int64(math.MaxInt64), controllerInstanceBase.sortedControllerInstancesLocal[0].controllerKey)
 }
 
-func TestGetMaxInterval(t *testing.T) {
+func TestGenerateKey(t *testing.T) {
+	controllerBase := new(ControllerBase)
+
+	// Start state: []
+	// End state: [100]
+	// controllerKey: math.MaxInt64 * 100%
+	// Description: Start state [] means there is no controller. End state [100] means the new controller will
+	//              controll 100% of the scope space. The total scope space is math.MaxInt64 (9223372036854775807).
+	//              In this way, we expect controller key is math.MaxInt64.
+	//              (Same expression logic for other test cases)
+	controllerKey := GenerateKey(controllerBase)
+	assert.Equal(t, int64(9223372036854775807), controllerKey)
+
+	// Start state: [100]
+	// End state: [50, 50]
+	// controllerKey: math.MaxInt64 * 50% -- Scope of 100% splitted
+	controllerBase.sortedControllerInstancesLocal = []controllerInstanceLocal{
+		{
+			lowerboundKey: 0,
+			controllerKey: int64(9223372036854775807),
+		},
+	}
+	controllerKey = GenerateKey(controllerBase)
+	assert.Equal(t, int64(4611686018427387903), controllerKey)
+
+	// Start state: [50, 50]
+	// End state: [25, 25, 50]
+	// controllerKey: math.MaxInt64 * 25%  -- Scope of first 50% splitted
+	controllerBase.sortedControllerInstancesLocal = []controllerInstanceLocal{
+		{
+			lowerboundKey: 0,
+			controllerKey: int64(4611686018427387903),
+		},
+		{
+			lowerboundKey: int64(4611686018427387903),
+			controllerKey: int64(9223372036854775807),
+		},
+	}
+	controllerKey = GenerateKey(controllerBase)
+	assert.Equal(t, int64(2305843009213693951), controllerKey)
```

Comment:
Make int64(2305843009213693951) and int64(4611686018427387903) const

Reply:
Updated.